proto/tendermint/liquidity/v1beta1/tx.proto

```diff
@@ -0,0 +1,207 @@
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"1\"",
+      format: "uint32"
+  }];
+
+  // reserve coin pair of the pool to deposit
+  repeated cosmos.base.v1beta1.Coin deposit_coins = 4 [(gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"deposit_coins\"",
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "[{\"denom\": \"denomX\", \"amount\": \"1000000\"}, {\"denom\": \"denomY\", \"amount\": \"2000000\"}]",
+      format: "sdk.Coins"
+    }];
+}
+
+// MsgCreatePoolResponse defines the Msg/CreatePool response type.
+message MsgCreatePoolResponse {}
+
+// `MsgDepositWithinBatch defines` an `sdk.Msg` type that supports submitting deposit request to the batch of the liquidity pool
+// Deposit submit to the batch of the Liquidity pool with the specified `pool_id`, `deposit_coins` for reserve
+// this requests are stacked in the batch of the liquidity pool, not immediately processed and
+// processed in the `endblock` at once with other requests.
+//
+// See: https://github.com/tendermint/liquidity/blob/develop/x/liquidity/spec/04_messages.md
+message MsgDepositWithinBatch {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string depositor_address = 1 [(gogoproto.moretags) = "yaml:\"depositor_address\"",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "account address of the origin of this message",
+      example: "\"cosmos1e35y69rhrt7y4yce5l5u73sjnxu0l33wvznyun\"",
+      format: "sdk.AccAddress"
+    }];
+
+  // id of the target pool
+  uint64 pool_id = 2 [(gogoproto.moretags) = "yaml:\"pool_id\"", (gogoproto.jsontag) = "pool_id",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"1\"",
+      format: "uint64"
+    }];
+
+  // reserve coin pair of the pool to deposit
+  repeated cosmos.base.v1beta1.Coin deposit_coins = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"deposit_coins\"",
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "[{\"denom\": \"denomX\", \"amount\": \"1000000\"}, {\"denom\": \"denomY\", \"amount\": \"2000000\"}]",
+      format: "sdk.Coins"
+    }];
+
+}
+
+// MsgDepositWithinBatchResponse defines the Msg/DepositWithinBatch response type.
+message MsgDepositWithinBatchResponse {}
+
+// `MsgWithdrawWithinBatch` defines an `sdk.Msg` type that supports submitting withdraw request to the batch of the liquidity pool
+// Withdraw submit to the batch from the Liquidity pool with the specified `pool_id`, `pool_coin` of the pool
+// this requests are stacked in the batch of the liquidity pool, not immediately processed and
+// processed in the `endblock` at once with other requests.
```

Comment:
```suggestion
// Submit a withdraw request to the liquidity pool batch with the specified `pool_id`, `pool_coin`.
// This request is added to the pool and executed at the end of the batch (`endblock`).
```

Reply:
if this description is accurate (I'm trying to understand, please forgive assumptions!) then we can rewrite the other message descriptions in these proto files and in the docs

Reply:
All requests (deposit, withdraw, swap) are accumulated in what is called batch and they are processed at the end of each block. 

In Cosmos SDK, there are `BeginBlocker` and `EndBlocker`. They will be triggered at the beginning and at the end of each block respectively. In this liquidity module, we add [ExecutePoolBatch](https://github.com/tendermint/liquidity/blob/develop/x/liquidity/abci.go#L22-L25) logic to execute whatever requests (deposit, withdraw, swap) are accumulated in batch. I hope this makes sense? If you need more explanation, let me know :)

Reply:
Very good explanation, I think the suggested change still makes sense.

Reply:
@defihyung are BeginBlocker and EndBlocker the operations that begin and end a block? 
![image](https://user-images.githubusercontent.com/10067215/114234865-5c007f00-994d-11eb-98db-4f73b25d0906.png)

per https://github.com/cosmos/cosmos-sdk/blob/master/docs/spec/SPEC-SPEC.md 

If I understand correctly, we now have a new operation category for batch. Would it make sense to put the batch operations in a new file XX_batch.md?